Makefile.docker

```diff
@@ -18,6 +18,12 @@ docker-image: clean docker-image-no-clean docker-plugin-image docker-hubble-rela
 	$(MAKE) docker-operator-azure-image
 	$(MAKE) docker-operator-generic-image
 
+docker-images-no-clean: docker-image-no-clean docker-plugin-image docker-hubble-relay-image
```

Comment:
nit: should this be `docker-image-no-clean` instead of `docker-images-no-clean` just to be consistent?

Reply:
Looks like `docker-image-no-clean` is already another target as specified in the dependencies to this new target.

I agree though this is super easy to miss, should we name it `docker-images-all-no-clean` to help highlight the difference?

Reply:
@nebril @joestringer IMO we should clean up and not build all images from a singular target. Also, the build system should not require "cleaning" to produce correct images, so I'd like to separate cleaning out from the image targets.

How about:
docker-images-all: build all images
docker-image-agent: build only the Cilium agent image
docker-image: removed
docket-image-no-clean: removed

Reply:
Resolving this with:

- `docker-image-no-clean` -> `docker-cilium-image` (build cilium docker image)
  - `docker-image-no-clean-unstripped` -> `docker-cilium-image-unstripped` (build cilium docker image, unstripped)
- `docker-image` -> `docker-images-all` (build all docker images, no clean)
  - `docker-image-unstripped` -> `docker-images-all-unstripped` (build all docker images, no clean, unstripped)

- new targets: `docker-operator-images-all` and `docker-operator-images-all-unstripped`